app/src/full/java/io/homeassistant/companion/android/location/HighAccuracyLocationService.kt

```diff
@@ -159,26 +161,41 @@ class HighAccuracyLocationService : Service() {
 
     private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
 
+    override fun onCreate() {
+        notificationId = HIGH_ACCURACY_LOCATION_NOTIFICATION_ID.hashCode()
+
+        notificationManagerCompat = NotificationManagerCompat.from(this)
+
+        // Create notification
+        createNotificationBuilder(this)
+        notification = notificationBuilder.build()
+
+        super.onCreate()
+    }
+
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
         super.onStartCommand(intent, flags, startId)
 
         restartInProcess = false
         isRunning = true
         val intervalInSeconds = intent?.getIntExtra("intervalInSeconds", DEFAULT_UPDATE_INTERVAL_SECONDS) ?: DEFAULT_UPDATE_INTERVAL_SECONDS
         requestLocationUpdates(intervalInSeconds)
-        val notificationId = HIGH_ACCURACY_LOCATION_NOTIFICATION_ID.hashCode()
-        createNotificationBuilder(this)
-        startForeground(notificationId, notificationBuilder.build())
+
+        // Start service in foreground
+        startForeground(notificationId, notification)
         Log.d(TAG, "High accuracy location service (Interval: ${intervalInSeconds}s) started")
         return START_NOT_STICKY
     }
 
     override fun onDestroy() {
         fusedLocationProviderClient.removeLocationUpdates(getLocationUpdateIntent())
-        stopForeground(true)
+        ServiceCompat.stopForeground(this, ServiceCompat.STOP_FOREGROUND_REMOVE)
         isRunning = false
         Log.d(TAG, "High accuracy location service stopped")
 
+        // Remove notification again. Sometimes stopForeground(true) is not enough. Just to be sure
+        notificationManagerCompat.cancel(notificationId)
```

Comment:
I noticed I had to add similar logic in #1265 we just always have to cancel.  I think its due to the app being silently killed in the background.  I found a good way to test it is once you have the notification posted to remove the app from the recent apps list which isn't like a force stop but causes the variables to reset.  I think thats the case and reason for just always removing at the end.

Reply:
Thanks for help 👍  but also with force close app i couldn't reproduce the issue. It is weird. I can only  see this issue when i'm using this in my car. And also not every time :/ Most of the time it does work.